js/main.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+(function () {
+  var OFFERS_NUMBER = 8;
+  var OFFER_TITLES = ['Шикарная квартира', 'Уютная комната', 'Огромный дворец', 'Просторное бунгало'];
+  var HOUSE_TYPES = ['palace', 'flat', 'house', 'bungalo'];
+  var MAX_ROOMS = 10;
+  var MAX_GUESTS = 15;
+  var CHECK_INS = ['12:00', '13:00', '14:00'];
+  var CHECK_OUTS = ['12:00', '13:00', '14:00'];
+  var FEATURES = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+  var DESCRIPTIONS = ['Шикарная квартира', 'Уютная комната', 'Огромный дворец', 'Просторное бунгало'];
+  var PHOTOS = ['http://o0.github.io/assets/images/tokyo/hotel1.jpg', 'http://o0.github.io/assets/images/tokyo/hotel2.jpg', 'http://o0.github.io/assets/images/tokyo/hotel3.jpg'];
+  var MIN_COORDINATE_Y = 130;
+  var MAX_COORDINATE_Y = 630;
+
+  // Скрываем лэйаут карты
+  var map = document.querySelector('.map');
+  map.classList.remove('map--faded');
+  var mapPinsArea = document.querySelector('.map__pins');
+  var pinTemplate = document.querySelector('#pin').content.querySelector('.map__pin');
+
+  // Выводим рандомное число
+  // Добавляем +1 т.к. Math.floor округляет вниз, а Math.random(max) = 0,9 в периоде.
+  function getRandomNumber(maxNumber) {
+    return Math.floor(Math.random() * maxNumber + 1);
+  }
+
+  // Выводим рандомный элемент в массиве. Для этого возвращаем элемент с i = 0 до i = array.length - 1;
+  // Т.к. Единица не входит в рандом, а Math.random(max) = 0,9 в периоде.
+  function getRandomElementFromArray(array) {
+    return array[getRandomNumber(array.length) - 1];
+  }
+
+  // Получаем массив рандомной длины через Array.Of
+  // Шаффлим дефолтный массив, чтобы значения перемешивались в каждой итерации
+  // Возвращаем новый массив, начиная с первого элемента и до элемента с индексом emptyArray
+  function getRandomNumberOfElementsFromArray(arr) {
+    var emptyArrayWithRandomLenght = Array.of(getRandomNumber(arr.length));
```

Comment:
```suggestion
    var emptyArrayWithRandomLength = Array.of(getRandomNumber(arr.length));
```
интересно, у тебя же из-за этой опечатки код не так работал

Reply:
Орфографическая ошибка, переменная вызывается только на следующей строке

Reply:
орфографическая ошибка делает эти две переменные разными, типа mn и aa - абсолютно разные переменные. Так что в след строке там андефайн как значение переменной